### ISSUE TITLE
Graph Designer JS locks up

### ISSUE THREAD

**Issue by azaroth42** [archesproject/arches#4149]
When editing graphs, the javascript will very frequently lock up and cause the page to freeze.
See: http://recordit.co/8UMyu10OvW

Adding a relatively small branch to a small model shouldn't eventually get to the page is unresponsive browser error! 

Note that the response to the POST to update the model comes back very quickly.

**Comment by azaroth42** [comment]
Again, with the MacOS activity monitor running: http://recordit.co/cFrlddXZmB 

It pins the CPU at 100%, hence the unresponsive page issue rather than just incremental slowing down from running out of memory (per #3993).

**Comment by azaroth42** [comment]
One more: http://recordit.co/HpkK5lgAMu

**Comment by robgaston** [comment]
@azaroth42 I just merged in a PR that seems to fix the issues that you were seeing here - I was able to add a few dozen nodes to a graph w/o refreshing now and all seems well.

The issue was some javascript that was executing multiple times per save action and was ballooning out as the graph grew (for example, the D3 graph representation was being rerendered many times [not an inexpensive operation] just to add a single node).  I've managed to reduce those calls to once per request/response which seems to prevent the issue w/ the browser locking up as you were experiencing.

I think that we may still have some work to do to improve garbage collection/memory management, but I'm no longer experiencing any noticeable, serious performance problems when using the graph designer.

These changes are in `master` now and will be included in the imminent 4.3.2 release
